src/lib.rs

```diff
@@ -125,15 +149,67 @@ macro_rules! impl_argminmax {
+                        }
+                    }
+                    #[cfg(target_arch = "aarch64")]
+                    {
+                        if std::arch::is_aarch64_feature_detected!("neon") & (<$t>::NB_BITS < 64) {
+                            // We miss some NEON instructions for 64-bit numbers
+                            return unsafe { NEON::argminmax(self) }
+                        }
+                    }
+                    #[cfg(target_arch = "arm")]
+                    {
+                        if std::arch::is_arm_feature_detected!("neon") & (<$t>::NB_BITS < 64) {
+                            // TODO: requires v7?
+                            // We miss some NEON instructions for 64-bit numbers
+                            return unsafe { NEON::argminmax(self) }
+                        }
+                    }
+                    SCALAR::argminmax(self)
+                }
+
+                // As there are no NaNs when NOT using floats -> just use argminmax
+                fn nanargminmax(&self) -> (usize, usize) {
+                    self.argminmax()
+                }
+            }
+        )*
+    };
+}
+
+// Macro for implementing ArgMinMax for floats (f32, f64)
+macro_rules! impl_argminmax_float {
+    ($($t:ty),*) => {
+        $(
+            impl ArgMinMax for &[$t] {
+                fn nanargminmax(&self) -> (usize, usize) {
+                    #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
+                    {
+                        if is_x86_feature_detected!("sse4.1") & (<$t>::NB_BITS == 8) {
+                            // 8-bit numbers are best handled by SSE4.1
+                            return unsafe { SSE::argminmax(self) }
+                        } else if is_x86_feature_detected!("avx512bw") & (<$t>::NB_BITS <= 16) {
+                            // BW (ByteWord) instructions are needed for 8 or 16-bit avx512
+                            return unsafe { AVX512::argminmax(self) }
+                        } else if is_x86_feature_detected!("avx512f") {  // TODO: check if avx512bw is included in avx512f
+                            return unsafe { AVX512::argminmax(self) }
+                        } else if is_x86_feature_detected!("avx2") {
                             return unsafe { AVX2::argminmax(self) }
                         // SKIP SSE4.2 bc scalar is faster or equivalent for 64 bit numbers
                         // // } else if is_x86_feature_detected!("sse4.2") & (<$t>::NB_BITS == 64) & (<$t>::IS_FLOAT == false) {
                         //     // SSE4.2 is needed for comparing 64-bit integers
                         //     return unsafe { SSE::argminmax(self) }
                         } else if is_x86_feature_detected!("sse4.1") & (<$t>::NB_BITS < 64) {
                             // Scalar is faster for 64-bit numbers
+                            // TODO: double check this (observed different things for new float implementation)
```

Comment:
The `/benches/results` indicate that for most CPUs this is indeed faster! => will look into this in a separate PR